blockstore/splitstore/splitstore.go

```diff
@@ -618,13 +620,27 @@ func (s *SplitStore) trackTxnRefMany(cids []cid.Cid) error {
 	return nil
 }
 
+// This could get _real_ slow (many minutes to hours) and will block tipset verification. Can we do this lazily and/or force
+// the GC thread to handle this?
+//
+// I.e., take a lock to make sure GC is paused, then stick this in a "to traverse" queue. Then, when
+// GC "resumes", it first checks that queue.
+
 // transactionally protect a reference by walking the object and marking.
 // concurrent markings are short circuited by checking the markset.
 func (s *SplitStore) doTxnProtect(root cid.Cid, batch map[cid.Cid]struct{}) error {
 	// Note: cold objects are deleted heaviest first, so the consituents of an object
 	// cannot be deleted before the object itself.
 	err := s.walkObjectIncomplete(root, cid.NewSet(),
 		func(c cid.Cid) error {
+			// Let's make sure we _completely_ match vm.Copy. We don't need this _now_,
+			// but should try to not to introduce inconsistencies.
+
+			if root.Prefix().MhType == 0 {
+				// identity cid, skip
+				return errStopWalk
+			}
```

Comment:
let's move this into `isFilCommitment` and rename that function, so that we don't have to sprinkle it everywhere.

Reply:
done in https://github.com/filecoin-project/lotus/pull/6474/commits/de5e21bf1a86746166ee9cb637d4121ccde10bba
I also added code to handle identity cids at the interface boundary so as to be consistent with wrapped idstores.